tests/lib/rules/no-extra-parens.js

```diff
@@ -706,6 +767,43 @@ ruleTester.run("no-extra-parens", rule, {
         invalid("foo instanceof (bar)", "foo instanceof bar", "Identifier", 1, { options: ["all", { nestedBinaryExpressions: false }] }),
         invalid("foo in (bar)", "foo in bar", "Identifier", 1, { options: ["all", { nestedBinaryExpressions: false }] }),
         invalid("foo + (bar)", "foo + bar", "Identifier", 1, { options: ["all", { nestedBinaryExpressions: false }] }),
-        invalid("foo && (bar)", "foo && bar", "Identifier", 1, { options: ["all", { nestedBinaryExpressions: false }] })
+        invalid("foo && (bar)", "foo && bar", "Identifier", 1, { options: ["all", { nestedBinaryExpressions: false }] }),
+
+        // ignoreJSX === "multi-line"
+        invalid("const Component = (<div />);", "const Component = <div />;", "JSXElement", 1, {
+            options: ["all", { ignoreJSX: "multi-line" }]
+        }),
+        invalid([
+            "const Component = (",
+            "  <div />",
+            ");"
+        ].join("\n"), "const Component = \n  <div />\n;", "JSXElement", 1, {
+            options: ["all", { ignoreJSX: "multi-line" }]
+        }),
+
+        // ignoreJSX === "single-line"
+        invalid([
+            "const Component = (",
+            "<div>",
+            "  <p />",
+            "</div>",
+            ");"
+        ].join("\n"), "const Component = \n<div>\n  <p />\n</div>\n;", "JSXElement", 1, {
+            options: ["all", { ignoreJSX: "single-line" }]
+        }),
+        invalid([
+            "const Component = (<div>",
+            "  <p />",
+            "</div>);"
+        ].join("\n"), "const Component = <div>\n  <p />\n</div>;", "JSXElement", 1, {
+            options: ["all", { ignoreJSX: "single-line" }]
+        }),
+        invalid([
+            "const Component = (<div",
+            "  prop={true}",
+            "/>)"
+        ].join("\n"), "const Component = <div\n  prop={true}\n/>", "JSXElement", 1, {
+            options: ["all", { ignoreJSX: "single-line" }]
+        })
```

Comment:
Could you add a test case for `{ ignoreJSX: "none" }`? It would be good to make sure that this behaves as expected.